### PR TITLE
PUB-1917 - Readme changes

### PR DESCRIPTION
**Pull request by furnivall**
### JIRA link
https://tools.hmcts.net/jira/browse/PUB-1917

### Change description
Added readme. There are three environment variables that I don't understand. Please amend them accordingly!

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by joshblackmoor — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -1,233 +1,249 @@
...
+
+## Overview
+`pip-frontend` is a microservice that serves the frontend of the Court and Tribunal Hearings Service (known as CaTH hereafter). This project was formerly known as the Publications and Information Project within HMCTS.
+
+The frontend uses a [Node.js](https://nodejs.org/en) runtime environment, using [Express.js](https://expressjs.com/) as the web application framework.
+Our templating engine is [nunjucks](https://mozilla.github.io/nunjucks/). We mostly use [TypeScript](https://www.typescriptlang.org/) for application code.
+
+It is connected to several other microservices in production (all written in Java/Spring Boot):
+| Microservice  | Summary |
+| ------------- | ------------- |
```
Comment (KianKwa) [inline README.md:45]: Something wrong with the syntax for table here. I pulled down the readme file and the table is not rendering correctly.

File: README.md [hunk 2]
```diff
@@ -1,233 +1,249 @@
...
-
-The Publishing & Information Hub will be responsible for the following:
-
--   Receiving data from source systems, such as Schedule & Listing, via Hearings Management Interface
--   Publish lists, outcomes, judgements and management information onto GOV.UK
--   Provide functionality to display information in court and tribunals buildings on the relevant hardware
--   Comply with Open Justice procedures and business rules
+# pip-frontend
+![Light Mode Logo](lightLogo.png#gh-light-mode-only)
+![Dark Mode Logo](darkLogo.jpg#gh-dark-mode-only)
```
Comment (KianKwa) [inline README.md:3]: The logo is showing weirdly on my screen. Can see the outline of the light mode logo and also the dark mode logo. Not sure if we need the logo at all.

image

File: README.md [hunk 3]
```diff
@@ -1,233 +1,249 @@
...
+| REDIS_LOCAL                      | Boolean to determine if Redis runs locally or not                                                                                                                                       |No|
+| TENANT_ID                        | Directory unique ID assigned to our Azure AD tenant. Represents the organisation that owns and manages the Azure AD instance.                                                           |No|
+| DATA_MANAGEMENT_AZ_API           | Used as part of the `scope` parameter when requesting a token from Azure. Used for service-to-service communication with the pip-data-management service                                |No|
+| ACCOUNT_MANAGEMENT_AZ_API        | Used as part of the `scope` parameter when requesting a token from Azure. Used for service-to-service communication with the pip-account-management service                             |No|
+| SUBSCRIPTION_MANAGEMENT_AZ_API   | Used as part of the `scope` parameter when requesting a token from Azure. Used for service-to-service communication with the pip-subscription-management service                        |No|
```
Comment (KianKwa) [inline README.md:137]: Think it is missing CHANNEL_MANAGEMENT_AZ_API

File: README.md [hunk 4]
```diff
@@ -1,233 +1,249 @@
...
+| B2C_USERNAME                | User's username for B2C authentication     |
+| B2C_PASSWORD                | User's password for B2C authentication     |
+| B2C_ADMIN_USERNAME          | B2C administrator's username               |
+| B2C_ADMIN_PASSWORD          | B2C administrator's password               |
+| CFT_INVALID_USERNAME        | Invalid username for CFT authentication    |
+| CFT_VALID_USERNAME          | Valid username for CFT authentication      |
+| CFT_VALID_PASSWORD          | Valid password for CFT authentication      |
+| CFT_INVALID_PASSWORD        | Invalid password for CFT authentication    |
+| B2C_SYSTEM_ADMIN_USERNAME   | B2C system administrator's username        |
+| B2C_SYSTEM_ADMIN_PASSWORD   | B2C system administrator's password        |
```
Comment (KianKwa) [inline README.md:156]: Might be worth mentioning TEST_URL and TEST_HEADLESS

Comment (KianKwa) [inline README.md:156]: For the new codecept end2end tests, will also need the following env variables:

DATA_MANAGEMENT_URL
SUBSCRIPTION_MANAGEMENT_URL
SYSTEM_ADMIN_PROVENANCE_ID
VERIFIED_USER_ID

File: README.md [hunk 5]
```diff
@@ -1,233 +1,249 @@
...
-Here's an example setup:
+#### Running e2e tests:
+Our legacy e2e tests can be run using:
 
-```json
-    "security": {
-      "referrerPolicy": "origin",
-    }
-```
+`yarn test:functional` (headless) or `yarn test:functional-dev`
```
Comment (KianKwa) [inline README.md:238]: old legacy test only should be run using 'yarn test:functional:old'. The one mentioned here `yarn test:functional' will run both old and new tests

File: README.md [hunk 6]
```diff
@@ -1,233 +1,249 @@
...
 
-Make sure you have those values set correctly for your application.
+New tests can be run using:
 
-### Healthcheck
+`yarn test:functional:all`
```
Comment (KianKwa) [inline README.md:242]: Might also worth mentioning new tests can be run using both 'yarn test:functional:all' (both subset tests for PR and nightly tests) and 'yarn test:functional:min' (only subset tests)

File: README.md (hunk 1 above)
Comment (ChrisS1512) [reply README.md:46]: Fixed :)

File: README.md (hunk 2 above)
Comment (ChrisS1512) [reply README.md:3]: I had the same issue, however it only seems to happen locally on the machine. In github all seems OK. Could you check in github to see if it's the same for you?

File: README.md [hunk 7]
```diff
@@ -1,233 +1,249 @@
...
+|pip-publication-services(`port:8081`)|Handles operations related to sending of notification emails to verified users, admin users and publication subscribers using [GOV.UK Notify](https://www.notifications.service.gov.uk/), as well as forwarding of publications to third-party publishers.|
+|pip-account-management(`port:6969`)|Handles operations related to accounts, including interaction with Azure B2C for PI_AAD users. It also manages the audit functionality.|
+
+Most of the communication with this service benefits from using secure authentication. While possible to stand up locally in insecure mode, to simulate a production environment it is better to use secure mode.
+
+
+### Features and functionality
+- Viewable by users directly in either HTTP or HTTPS mode (default port: 8080)
+- Uploading of publication files using a web interface within the [manual upload](./src/main/views/manual-upload.njk) view.
+- Account setup, sign-in and user management functionality. Sign-in and password management is managed using Azure B2C user flows.
```
[the remark this answers is not in the document]
Comment (ChrisS1512) [reply README.md]: Added

File: README.md [hunk 8]
```diff
@@ -1,233 +1,249 @@
...
+- View publications directly in the browser restricted to the user's account privileges.
+- Processes to manage creation of a new media account for a user with administrator oversight (approve/reject).
+- Tiered access to specific functionality and content within three main categories (media/administrator/system administrator), as well as unauthenticated functionality.
+- Management functionality for a maximum of 4 system administrators (set by environment variable). System admins are able to see audit actions by regular administrators, view underlying data, manage users etc.
+- Set up subscriptions to be notified via email when a new publication with given parameters is uploaded.
+- Includes a large collection of [custom nunjucks filters](./src/main/modules/nunjucks/njkFilters.ts) used for specific functionality within the application.
```
[the remark this answers is not in the document]
Comment (ChrisS1512) [reply README.md]: I've moved this to a separate section

File: README.md (hunk 3 above)
Comment (ChrisS1512) [reply README.md]: Added

File: README.md (hunk 4 above)
Comment (ChrisS1512) [reply README.md]: Added

File: README.md (hunk 6 above)
Comment (ChrisS1512) [reply README.md]: Added

**Review by KianKwa — APPROVED** [review]

**Review by joshblackmoor — APPROVED** [review]

**Pull request merged**